src/execution/execute.js

```diff
@@ -916,6 +918,58 @@ function completeValue(
   );
 }
 
+/**
+ * Complete a async iterable value by completing each item in the list with
+ * the inner type
+ */
+
+function completeAsyncIterableValue(
+  exeContext: ExecutionContext,
+  returnType: GraphQLList<GraphQLOutputType>,
+  fieldNodes: $ReadOnlyArray<FieldNode>,
+  info: GraphQLResolveInfo,
+  path: Path,
+  result: AsyncIterable<mixed>,
+): Promise<$ReadOnlyArray<mixed>> {
+  // $FlowFixMe
+  const iteratorMethod = result[SYMBOL_ASYNC_ITERATOR];
+  const iterator = iteratorMethod.call(result);
+
+  const completedResults = [];
+  let index = 0;
+
+  const itemType = returnType.ofType;
+
+  const handleNext = () => {
+    const fieldPath = addPath(path, index);
+    return iterator.next().then(
+      ({ value, done }) => {
+        if (done) {
+          return;
```

Comment:
You can return `completedResults` here right?
And save one tick.

Reply:
Yes, updated to return `completedResults` here and in the error handler.